src/mrdog.rs

```diff
@@ -1,19 +1,33 @@
+use std::error::Error;
+
+use clap::ValueEnum;
+use serde::{Deserialize, Serialize};
+
 use self::change_request::{ChangeRequest, ChangeRequestProvider};
 
 mod change_request;
 mod config_storage;
 
-#[derive(Debug)]
-pub enum ConfigItem {
-    GitLabApiToken(String),
-    GitHubApiToken(String),
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum, Deserialize, Serialize)]
+pub enum Provider {
+    Gitlab,
+    Github,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ConfigValue {
+    pub name: String,
+    pub value: String,
+    pub provider: Provider,
 }
 
-pub fn set_config(item: ConfigItem) {
-    config_storage::set_config(item)
+pub fn set_config(item: ConfigValue) -> Result<(), Box<dyn Error>> {
```

Comment:
NIT: Ideally, instead of dyn/sad we'd have specific error enum: FileNotFound, InvalidFormat, etc.